### PR TITLE
New loss

### PR DESCRIPTION
**Pull request by tisabe**
@speckhard Add binary cross entropy loss to train.py and options for changing loss in config. Also add tests for new loss function.

### CODE REVIEW

File: jraph_MPEU/train.py [hunk 1]
```diff
@@ -407,6 +408,45 @@ def create_optimizer(
...
+
+def loss_fn_bce(params, state, rng, graphs, net_apply):
+    """Binary cross entropy loss function for classification."""
+    labels = graphs.globals
+    graphs = replace_globals(graphs)
+    targets = jax.nn.one_hot(labels, 2)
+
+    # try get_valid_mask function instead
+    mask = jraph.get_graph_padding_mask(graphs)
+    pred_graphs, new_state = net_apply(params, state, rng, graphs)
```
Comment (speckhard) [inline jraph_MPEU/train.py:438]: It looks like from your test the new_state=state when we call net_apply. This seems to match with haiku docs: where they say the most common usage of state is to `represent the moving averages used in batch normalization`. So for us this variable is not used since we're not doing batch norm right?
